### PR TITLE
Add directories to the file blacklist

### PR DESCRIPTION
**Pull request by kamijean**

### CODE REVIEW

**Comment by nick-benoit14** [comment]
Looks good to me.

File: lib/blacklight/xml_parser.rb [hunk 1]
```diff
@@ -61,14 +61,15 @@ def self.iterate_xml(resources, zip_file)
...
+    FILE_BLACK_LIST.any? { |black_item| File.fnmatch?(black_item, name) } ||
+      type == :directory
   end
 
-  def self.iterate_files(zip_file)
-    zip_file.
-      entries.select { |e| !black_listed?(e.name) }.
-      map { |entry| BlacklightFile.new(entry) }
+  def self.iterate_files(zipfile)
+    zipfile.entries.select { |e| !black_list?(e.name, e.ftype) }.map do |entry|
```
Comment (bfcoder) [inline lib/blacklight/xml_parser.rb]: why this refactor? it looked much cleaner multilined the way it was

Comment (bfcoder) [reply lib/blacklight/xml_parser.rb]: I think you should revert this back to the way it was and add the `e.ftype` as needed

File: lib/blacklight/xml_parser.rb [hunk 2]
```diff
@@ -61,13 +61,14 @@ def self.iterate_xml(resources, zip_file)
...
+  def self.black_list?(name, type)
+    FILE_BLACK_LIST.any? { |black_item| File.fnmatch?(black_item, name) } ||
+      type == :directory
   end
 
-  def self.iterate_files(zip_file)
-    zip_file.
-      entries.select { |e| !black_listed?(e.name) }.
+  def self.iterate_files(zipfile)
+    zipfile.
```
Comment (bfcoder) [inline lib/blacklight/xml_parser.rb]: Don't hate me, but one more thing. :] Can you move entries up to this line? Keep the select and map on separate lines?
```ruby
zipfile.entries.
  select {}.
  map {}
```
😁

File: lib/blacklight/xml_parser.rb [hunk 3]
```diff
@@ -61,13 +61,14 @@ def self.iterate_xml(resources, zip_file)
     resources_array.flatten - ["", nil]
   end
 
-  def self.black_listed?(name)
-    FILE_BLACK_LIST.any? { |b_list_item| File.fnmatch?(b_list_item, name) }
+  def self.black_list?(name, type)
+    FILE_BLACK_LIST.any? { |black_item| File.fnmatch?(black_item, name) } ||
+      type == :directory
```
Comment (bfcoder) [inline lib/blacklight/xml_parser.rb:66]: In the interest of performance, should we do this check first?

Comment (bfcoder) [reply lib/blacklight/xml_parser.rb:66]: nevermind, it's fine the way it is.

**Comment by bfcoder** [comment]
Looks good.

**Pull request merged**